src/common/RootProviders.tsx

```diff
@@ -29,6 +30,7 @@ export function RootProviders({
                 <ActiveProjectContextProvider>
                   <SafeAreaProvider>
                     <NavigationContainer>{children}</NavigationContainer>
+                    <NoInternetToast />
```

Comment:
I see - so, in the case that we want to utilize `<Toast />` for other purposes that are also global, they can be siblings to this.  👍 I dig the simplicity.  If we someday want a `<Toast />` here that tries to serve all "global toast" needs, we can always refactor into that later.

Reply:
Or no, shoot - reading through the [docs](https://github.com/calintamas/react-native-toast-message/blob/945189fec9746b79d8b5b450e298ef391f8022fb/docs/modal-usage.md) it sounds like the `<Toast />` element is really designed to be globally reused.  I get this impression esp. from that modal page's notes about `refs` etc.:

> The ref is tracked automatically; whichever <Toast /> instance last had its ref set will be used when showing/hiding.

The [react-native-model example](https://github.com/calintamas/react-native-toast-message/blob/945189fec9746b79d8b5b450e298ef391f8022fb/docs/modal-usage.md#notes-regarding-react-native-modal-or-nativestacknavigator) similarly makes me think we'd never want _sibling_ `<Toast />` elements (the internal ref tracking would cause only the last one to be used).  This makes me think this line really should just be `<Toast />`, right?  I better understand now what you were saying on how `NoInternetToast` isn't really a component - it's really just the `useEffect` that we need.  I'd personally turn to a context Provider that just renders its children for this.

Reply:
Ahh, yes. Keeping the Toast and the NoInternetToast "effect" separate is a better path then. I've pushed another commit that I'm hopeful will be what we all want.